sec2_the-ruby-programming-language/2.7_arrays-iterators.rb

```diff
@@ -0,0 +1,91 @@
+# Arrays
+a = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+p a       # [1, 2, 3, 4, 5, 6, 7, 8, 9]
+p a.last  # 9
+
+# Range
+x = 1..10 # inclusive
+p x
+p x.class # Range (class type)
+p x.to_a          # [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+p x.to_a.shuffle  # [2, 7, 8, 1, 10, 4, 3, 9, 5, 6]
+p x               # 1..10
+p x.to_a.shuffle! # [8, 5, 7, 4, 1, 6, 10, 3, 9, 2]
+p x               # 1..10
+
+z = x.to_a.shuffle!
+p z               # [9, 10, 7, 1, 6, 8, 2, 3, 4, 5]
+
+x = (1..10).to_a
+p x               # [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+p x.reverse       # [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
+p x               # [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+p x.reverse!      # [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
+p x               # [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]
+
+
+x = "a".."d"
+p x.to_a            # ["a", "b", "c", "d"]
+p x.to_a.shuffle    # ["c", "d", "a", "b"]
+# p x.length        // undefined method `length' for "a".."d":Range (NoMethodError)
+y = x.to_a.shuffle
+p y                 # ["a", "b", "c", "d"]
+p y.length          # 4
+
+
+# Append
+a = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+p a.length
+a << 10
+p a                     # [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
```

Comment:
line 39, `<<` is appending a 10 into `a`, you can append anything (it does not limit to be `int` based on the case above)